### PR TITLE
Use add_loss in VAE examples

### PR DESCRIPTION
**Pull request by Spotlight0xff**
Hi!

I've implemented the `add_loss` feature in the example for the variational autoencoder (VAE), with both MLP and CNN.
But I'm not sure If I used it correctly, as there is little to no documentation on this feature as of now.
One thing which wonders me is the usage of `zero_loss` in the code, as shown in this [issue](https://github.com/fchollet/keras/issues/5563) (which basically does nothing).
It passes the PEP8 checks, I appreciate any concern towards the idiomatic use of the keras 2.0 API.

For reference, the HN discussion is [here](https://news.ycombinator.com/item?id=13873837).

edit: Fixed a typo in the vae_loss function.

Thanks for your time.
André

### CODE REVIEW

File: examples/variational_autoencoder.py [hunk 1]
```diff
@@ -41,13 +42,34 @@ def sampling(args):
...
+        x = inputs[0]
+        x_decoded_mean = inputs[1]
+        loss = self.vae_loss(x, x_decoded_mean)
+        self.add_loss(loss, inputs=inputs)
+        # we don't use this output, but it has to have the correct shape:
+        return K.ones_like(x)
+
+loss_layer = CustomVariationalLayer()([x, x_decoded_mean])
+vae = Model(x, [loss_layer])
+vae.compile(optimizer='rmsprop', loss=[zero_loss])
```
Comment (fchollet) [inline examples/variational_autoencoder.py]: If you don't need to create a loss dependent on model outputs, you can use `loss=None`. In that case only the loss you added with `add_loss` earlier will be used.

Comment (Spotlight0xff) [inline examples/variational_autoencoder.py]: Yes, I thought so too.
However, when I change `loss=[zero_loss]` to `loss=None` I get this error:

```python
/home/spotlight/git/keras/keras/engine/training.py:729: UserWarning: Output "custom_variational_layer_1" missing from loss dictionary. We assume this was done on purpose, and we will not be expecting any data to be passed to "custom_variational_layer_1" during training.
  '" during training.')
Traceback (most recent call last):
  File "variational_autoencoder.py", line 86, in <module>
    validation_data=(x_test, x_test))
  File "/home/spotlight/git/keras/keras/engine/training.py", line 1405, in fit
    batch_size=batch_size)
  File "/home/spotlight/git/keras/keras/engine/training.py", line 1299, in _standardize_user_data
    exception_prefix='model target')
  File "/home/spotlight/git/keras/keras/engine/training.py", line 100, in _standardize_input_data
    'Found: array with shape ' + str(data.shape))
ValueError: The model expects 0 input arrays, but only received one array. Found: array with shape (60000, 784)
```

Should I open an issue?

Comment (fchollet) [inline examples/variational_autoencoder.py]: This is the error that one would except. If there is not loss on the output, why would you pass any target data?

**Comment by fchollet** [comment]
Any update?

**Comment by fchollet** [comment]
No update in a while, so I fixed it and merged it.

**Pull request merged**